gammapy/estimators/utils.py

```diff
@@ -753,3 +754,79 @@ def get_combined_significance_maps(estimator, datasets):
         npred_excess=npred_excess_sum,
         estimator_results=results,
     )
+
+
+def joint_flux_maps(maps, method="gaussian_errors", **kwargs):
+    """Create a FluxMaps by combining a list of flux maps with the same geometry.
+     This assumes the flux maps are independent measurements of the same true value.
+
+    Parameters
+    ----------
+    maps : list of `~gammapy.estimators.FluxMaps` or ~gammapy.estimators.FluxMaps`
+        List of maps with the same geometry.
+    kwargs : dict
+        Options passed to `~gammapy.estimators.FluxMaps.from_maps`
+    method : {"gaussian_errors"}
+        * gaussian_errors :
+            Under the gaussian error approximation the likelihood is given by the gaussian distibution.
+            The product of gaussians is also a gaussian so can derive dnde, dnde_err, and ts.
+
+    Returns
+    -------
+    flux_maps : `~gammapy.estimators.FluxMaps`
+        Joint flux map.
+    """
+    if method != "gaussian_errors":
+        raise ValueError(
+            f'Invalid method : {method}, available methods are : "gaussian_errors"'
+        )
+
+    if isinstance(maps, FluxMaps):
+        geom = maps.dnde.sum_over_axes().geom
+        means = list(maps.dnde.copy().iter_by_image(keepdims=True))
+        sigmas = list(maps.dnde_err.copy().iter_by_image(keepdims=True))
+        mean = Map.from_geom(geom, data=means[0].data, unit=means[0].unit)
+        sigma = Map.from_geom(geom, data=sigmas[0].data, unit=sigmas[0].unit)
+        gti = maps.gti
+    else:
+        means = [map_.dnde.copy() for map_ in maps]
+        sigmas = [map_.dnde_err.copy() for map_ in maps]
+        mean = means[0]
+        sigma = sigmas[0]
+
+        gtis = [map_.gti for map_ in maps if map_.gti is not None]
+        if np.any(gtis):
+            gti = gtis[0].copy()
+            for k in range(1, len(gtis)):
+                gti.stack(gtis[k])
+        else:
+            gti = None
+
+    for k in range(1, len(means)):
+
+        mean_k = means[k].quantity.to_value(mean.unit)
+        sigma_k = sigmas[k].quantity.to_value(sigma.unit)
+
+        mask_valid = np.isfinite(mean) & np.isfinite(sigma) & (sigma.data != 0)
+        mask_valid_k = np.isfinite(mean_k) & np.isfinite(sigma_k) & (sigma_k != 0)
+        mask = mask_valid & mask_valid_k
+        mask_k = ~mask_valid & mask_valid_k
+
+        mean.data[mask] = (
+            (mean.data * sigma_k**2 + mean_k * sigma.data**2)
+            / (sigma.data**2 + sigma_k**2)
+        )[mask]
+        sigma.data[mask] = (
+            sigma.data * sigma_k / np.sqrt(sigma.data**2 + sigma_k**2)
+        )[mask]
+
+        mean.data[mask_k] = mean_k[mask_k]
+        sigma.data[mask_k] = sigma_k[mask_k]
+
+    ts = -2 * np.log(
+        stats.norm.pdf(0, loc=mean, scale=sigma)
+        / stats.norm.pdf(mean, loc=mean, scale=sigma)
+    )
+
+    kwargs["sed_type"] = "dnde"
+    return FluxMaps.from_maps(dict(dnde=mean, dnde_err=sigma, ts=ts), gti=gti, **kwargs)
```

Comment:
The `dict` should contain Maps, no? Right now it contains numpy arrays

Reply:
I added a line to redefine ts as a map, the others are already maps